butterfly-core/src/main/java/com/paypal/butterfly/core/UpgradePath.java

```diff
@@ -59,8 +60,12 @@
         }
         if (upgradeVersion == null || upgradeVersion.isEmpty()) {
             upgradeVersion = getLastVersion();
+        } else if(firstStep.getCurrentVersion().equals(upgradeVersion)){
+            throw new IllegalArgumentException("The requested upgrade version is the same as the version the application is currently at");
```

Comment:
Let's add the version in the exception message please. Like this:

`throw new IllegalArgumentException("The requested upgrade version (" + upgradeVersion + ") is the same as the version the application is currently at");`

Reply:
Fixed.